File: app/src/main/java/com/leilei/guoshujinfu/mylearning/util/BaseFragment.java
package com.leilei.guoshujinfu.mylearning.util;

import android.app.Fragment;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @Author: AqrLei
 * @Name MyLearning
 * @Description:
 * @Date: 2017/8/3
 */

public abstract class BaseFragment<T extends BaseActivity> extends Fragment {
    protected  T mContainerActivity;
    protected View mView;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if(context instanceof BaseActivity) {
            mContainerActivity = (T)context;
        }
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {

        return super.onCreateView(inflater, container, savedInstanceState);
    }
    protected abstract int getLayoutRes();

    protected void initComponents() {
    }

}

File: truckrental/src/main/kotlin/com/aqrlei/graduation/truckrental/baselib/base/BaseFragment.kt
package com.aqrlei.graduation.truckrental.baselib.base

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.aqrlei.graduation.truckrental.baselib.util.AppLog


/**
 * @Author: AqrLei
 *@Name MyLearning
 *@Description:
 *@Date: 2017/8/22
 */
abstract class BaseFragment<T : BaseActivity> : Fragment() {
    protected lateinit var mContainerActivity: T
    protected lateinit var mView: View


    protected abstract val layoutRes: Int
    protected open fun initComponents() {

    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is BaseActivity) {
            mContainerActivity = context as T
            AppLog.logDebug(AppLog.LOG_TAG_FRAGMENT, this.javaClass.simpleName + "Attach" +
                    mContainerActivity.javaClass.simpleName)

        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mView = inflater!!.inflate(layoutRes, container, false)
        initComponents()
        return mView
    }

    fun finish() {
        mContainerActivity.finish()
    }


}